src/podcast/feed_parser.py

```
@@ -0,0 +1,479 @@
+        # Parse the feed
+        feed = feedparser.parse(
+            feed_url,
+            agent=self.user_agent,
+        )
+
+        # Check for errors
+        if feed.bozo and feed.bozo_exception:
+            logger.warning(f"Feed parsing warning: {feed.bozo_exception}")
+
+        if not feed.feed:
+            raise ValueError(f"Failed to parse feed: {feed_url}")
```

Ruff (0.14.7)

Avoid specifying long messages outside the exception class

(TRY003)